### ISSUE TITLE
FeatureSpace multiple output from one input

### ISSUE THREAD

**Issue by zippeurfou** [keras-team/keras#19697]
This is more of an ask to add to a tutorial than a feature request as I believe this might already doable today.
When looking at the FeatureSpace tutorials it assumes that you create one output per feature.
I am basing this from [this](https://keras.io/examples/structured_data/feature_space_advanced/) tutorial where it seems to be an assumption that from one input you create one output.
For example:
```
feature_space = FeatureSpace(
    features={
        ...
        # Numerical features to normalize and bin
        "age": FeatureSpace.float_discretized(num_bins=4),
        ...
    },
  ...
    ],
    output_mode="concat",
)
```
In this example there is an assumption that you only create one float_discretized output for the input age.
However, in practice (eg. youtube [paper](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/45530.pdf)) multiple output can be created from one input. 
<img width="546" alt="Screenshot 2024-05-09 at 8 05 04 PM" src="https://github.com/keras-team/keras/assets/4520383/e9cba7bb-e60f-476f-acf5-bb4ce140e9e4">
It would be nice to add in the tutorial or somewhere in the doc an example to do so. I have tried to replicate the paper on numerical features using FeatureSpace and I found it difficult to do so without doing the extra transformation in the model itself which I think is not the initial idea of this functionality.

**Comment by fchollet** [comment]
This does not appear to require any extra features:

1. Use `FeatureSpace` to get `x`.
2. Call `keras.ops.square(x)` and `keras.ops.sqrt(x)` to get your other features.

> I found it difficult to do so without doing the extra transformation in the model itself

You can do the above either inside the model or in a data pipeline. Inside the model would look like this:

```python
# Retrieve a dict Keras Input objects
inputs = feature_space.get_inputs()
# Retrieve the corresponding encoded Keras tensors
encoded_features = feature_space.get_encoded_features()
x = encoded_features["x"]
x_sqrt = keras.ops.sqrt(x)
x_square = keras.ops.square(x)
output = ...
model = Model(inputs, output)
```

**Comment by zippeurfou** [comment]
Thank you @fchollet for the quick answer. The method you mentioned is how I do it already today.
I was hoping to be able to do it as part of the FeatureSpace creation (maybe wrongly?) just because it felt cleaner in term of code organization to me and still part of the feature creation to me. 
It also brings the question of combination within the FeatureSpace (eg. Normalization + x2).
So I guess the question is if it would make sense to extend FestureSpace to allows to have more flexibility than having a preprocessing transformation. In practice, if for example you send it a preprocessing that does Normalization and then x2 you would apply the adapt to normalization and then do x2

**Comment by zippeurfou** [comment]
Another more trivial example is for example if you want to transform one feature (eg. `age`) into:
1. Discretization that you will send after being discretized as cross with another feature.
2. `Normalization` that you won't cross
I don't think the current architecture allows you to do so but maybe I am missing something.
